pkg/virt-controller/watch/vmi/lifecycle.go

```diff
@@ -1086,3 +1099,16 @@ func preparePodPatch(oldPod, newPod *k8sv1.Pod) *patch.PatchSet {
 		patch.WithReplace("/status/conditions", newPod.Status.Conditions),
 	)
 }
+
+func handleEvictionInProgressAnnotation(vmi *virtv1.VirtualMachineInstance) *virtv1.VirtualMachineInstance {
+	const aValue = "kubevirt"
+	if vmi.IsMarkedForEviction() {
+		if vmi.Annotations == nil {
+			vmi.Annotations = map[string]string{}
+		}
+		vmi.Annotations[descheduler.EvictionInProgressAnnotation] = aValue
+	} else if vmi.Annotations != nil && vmi.Annotations[descheduler.EvictionInProgressAnnotation] != "" {
```

Comment:
```suggestion
	} else if _, ok := vmi.Annotations[descheduler.EvictionInProgressAnnotation]; ok {
```

That should work even if the map is nil? And will handle a set anno with empty value as well.